pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1362,7 +1362,7 @@ public ShutdownService getShutdownService() {
 
     protected String brokerUrl(ServiceConfiguration config) {
         if (config.getBrokerServicePort().isPresent()) {
-            return brokerUrl(ServiceConfigurationUtils.getAppliedAdvertisedAddress(config),
+            return brokerUrl(ServiceConfigurationUtils.getAppliedAdvertisedAddress(config, true),
```

Comment:
Why is the second parameter true? This fix has changed the default behavior. See `getAppliedAdvertisedAddress`:

```diff
 -       if (advertisedListener != null) {
 +       if (advertisedListener != null && !ignoreAdvertisedListener) {
```

Before this change, if `advertisedListener` is not null, it will go to this if block. However, after this change, it won't go to this if block because `!ignoreAdvertisedListener` is false. This may affect the Pulsar client's lookup result because `OwnershipCache#selfOwnerInfo` is set to the `PulsarService#brokerServiceUrl` in `OwnershipCache#refreshSelfOwnerInfo` method.

Reply:
Obtained from the commit record, the earliest logic is as follows
```
    public static String advertisedAddress(ServiceConfiguration config) {
        return ServiceConfigurationUtils.getDefaultOrConfiguredAddress(config.getAdvertisedAddress());
    }

    private String brokerUrl(ServiceConfiguration config) {
        if (config.getBrokerServicePort().isPresent()) {
            return brokerUrl(advertisedAddress(config), getBrokerListenPort().get());
        } else {
            return null;
        }
    }

    public static String brokerUrl(String host, int port) {
        return String.format("pulsar://%s:%d", host, port);
    }
```
The order of IP address acquisition is:  advertisedAddress, hostname 
And the port is get from `brokerServicePort` in ServiceConfiguration, not from advertisedListener. 
So there's no change in the original behavior
https://github.com/apache/pulsar/blob/05827aeb4b0aa4a1b09a7a122f00a65012bf53a1/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L415-L425

@BewareMyPower  Please correct me if I get this wrong

Reply:
It doesn't affect the bind address but affects `PulsarService#getClient` because it uses `brokerServiceUrl` as the service url. I'm not sure if it affected other things.

The lookup logic is my mistake and was not affected by this change.

Reply:
I tried to make it support advertisedListener before, but advertisedListener needs to use a key to get value. In this method, we don’t know what key to use

Reply:
No, the default key is the first key of `advertisedListeners`, see `validateAndAnalysisAdvertisedListener`. https://github.com/apache/pulsar/blob/05827aeb4b0aa4a1b09a7a122f00a65012bf53a1/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/validator/MultipleListenerValidator.java#L73-L75

Reply:
@hangc0276 PTAL since it's already cherry-picked to branch-2.8.

Reply:
@BewareMyPower do you want to create an issue and mark it "release/blocker" for   2.8.1 and 2.9.0 ?
it would be also better to advertise this problem on dev@pulsar

Reply:
Okay, just leave for a while, I'll send an email a few hours later.

Reply:
This PR break the behavior https://github.com/apache/pulsar/pull/10312

This PR fixes the previous issue  https://github.com/apache/pulsar/pull/10961

@BewareMyPower

Reply:
Got it. Thanks for your explanation. @315157973 

And I think there's no need to open a discussion anymore. @eolivelli